tests/test_crawl.py

```diff
@@ -277,3 +278,30 @@ def test_crawl_multiple(self):
 
         self._assert_retried(log)
         self.assertIn("Got response 200", str(log))
+
+    @defer.inlineCallbacks
+    def test_dns_server_ip_address(self):
+        from socket import gethostbyname
+
+        crawler = self.runner.create_crawler(SingleRequestSpider)
+        url = 'https://example.org'
+        yield crawler.crawl(seed=url)
```

Comment:
I’m not sure about making a request to https://example.org as part of the test suite. @kmike Thoughts?

Reply:
I don't like it either, and I've been told about this before, precisely by Mikhail 😄 (https://github.com/scrapy/scrapy/pull/2061#pullrequestreview-25525781)
But I don't know how else to check for an actual DNS resolution, the MockServer URLs uses only IP addresses.

Reply:
I don’t like this too, I’ll think what to do about it

Reply:
+1 not to use example.com, even if it means we need to figure out how to start a dummy DNS server in tests. It looks doable, as e.g. twisted has DNS server implementation (see https://twistedmatrix.com/documents/current/names/howto/custom-server.html)

Reply:
Indeed, I have a DNS forwarder running on a Raspberry Pi which is based on (should I say "copied from"?)  that page. I'll try to come up with a mock server. Thanks for the pointer.

Reply:
Upstream request removed 🙌